app/controllers/products_common_controller.rb

```diff
@@ -52,15 +52,16 @@ def show
   # GET /services/new
   def new
     @product = current_facility_products.new(account: Settings.accounts.product_default)
+    @product_form = EditProductForm.for(@product)
   end
 
   # POST /services
   def create
-    @product = current_facility_products.new(resource_params)
-    @product.initial_order_status_id = OrderStatus.default_order_status.id
-
-    if @product.save
-      flash[:notice] = "#{@product.class.name} was successfully created."
+    @product = current_facility_products.new(initial_order_status: OrderStatus.default_order_status)
+    @product_form = EditProductForm.for(@product)
+    @product_form.assign_attributes(resource_params)
+    if @product_form.save
```

Comment:
Maybe fold the `assign_attributes` and `save` into one `update(resource_params)` call for brevity?